File: c++/SelectionSort.cpp
#include <iostream>
using namespace std;



int main(){
    int arr[] = {4, 3, 6, 2, 1};
    int n = sizeof(arr) / sizeof(arr[0]);

    int min_index, temp;
    for (int i = 0; i < n; i++){

        min_index = i;
        for(int j = i+1; j < n; j++){
            if(arr[j] < arr[min_index])
                min_index = j;
        }

        temp = arr[min_index];
        arr[min_index] = arr[i];
        arr[i] = temp; 
    }

    for(int i =0; i < n; i++){
        cout << arr[i] << '\t';
    }
    cout << endl;
}

File: Python/SelectionSort.py
def selectionSort(arr, n):
    for i in range(0, n):
        min_index = i

        for j in range(i+1, n):
            if arr[j] < arr[min_index]:
                min_index = j

        arr[min_index], arr[i] = arr[i], arr[min_index]

    for i in arr:
        print(i, end='\t')

    print('')

arr = [3, 1, 2, 4, 6]
n = len(arr)

selectionSort(arr, n)